lang/java/tools/src/test/java/org/apache/avro/tool/TestDataFileTools.java

```diff
@@ -118,6 +118,33 @@ public void testReadToJsonPretty() throws Exception {
   }
 
   @Test
+  public void testReadHeadDefaultCount() throws Exception {
+    String expectedJson = jsonData.substring(0, 20); // first 10 numbers
+    assertEquals(expectedJson,
+      run(new DataFileReadTool(), "--head", sampleFile.getPath()));
+  }
+
+  @Test
+  public void testReadHeadEquals3Count() throws Exception {
+    String expectedJson = jsonData.substring(0, 6); // first 3 numbers
+    assertEquals(expectedJson,
+      run(new DataFileReadTool(), "--head=3", sampleFile.getPath()));
+  }
+
+  @Test
+  public void testReadHeadSpace5Count() throws Exception {
+    String expectedJson = jsonData.substring(0, 10); // first 5 numbers
+    assertEquals(expectedJson,
+      run(new DataFileReadTool(), "--head", "5", sampleFile.getPath()));
+  }
+
+  @Test
+  public void testReadHeadLongCount() throws Exception {
+    assertEquals(jsonData,
+      run(new DataFileReadTool(), "--head=3000000000", sampleFile.getPath()));
+  }
+
```

Comment:
with 0, I'd expect just no lines of output and success.

with negative, we can either go with posix ([it's an error](http://www.unix.com/man-page/posix/1p/head/)) or the gnu coreutils route and have negative mean "all but the last _n_ lines".

Personally, I'd favor aping posix here.